<p>While named function expressions might be useful for debugging purposes, some browsers do not support them correctly (for example Internet Explorer
8).</p>
<h2>Noncompliant Code Example</h2>
<pre>
f = function fun(){}; // Noncompliant;  named function expression
</pre>
<h2>Compliant Solution</h2>
<pre>
fun = function(){}; // Compliant; function expression
</pre>
<h2>Exceptions</h2>
<p>ECMAScript 6 generator functions are excluded from this rule. </p>
<pre>
function* f() {} // Compliant; generator function.
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

