<p>HTML-style comments are not part of EcmaScript specification, and should not be used.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;!-- Noncompliant --&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
// Compliant
/* Compliant */
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

