<p>Element selections that could be matched anywhere in the document can be very slow. That's why use of the universal selector, <code>*</code>,
should be limited; it explicitly specifies that the match could be anywhere.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$( ".buttons &gt; *" );  // Noncompliant; extremely expensive
</pre>
<h2>Compliant Solution</h2>
<pre>
$( ".buttons" ).children(); // Compliant
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

