<p>The assignment of default parameter values is generally intended to help the caller. But when a default assignment causes side effects, the caller
may not be aware of the extra changes or may not fully understand their implications. I.e. default assignments with side effects may end up hurting
the caller, and for that reason, they should be avoided.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var count = 0;

function go(i = count++) {  // Noncompliant
  console.log(i);
}

go();  // outputs 0
go(7); // outputs 7
go();  // outputs 1
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

