<p>
  读取对象中不存在的属性总是返回 <code>undefined</code>。
</p>

<h2>Noncompliant Code Example</h2>
<pre>
x = 42;
y = x.length;   // 不合规，数字类型没有 length 属性
</pre>
<h2>例外</h2>
<p>
  Ember 框架引入了一些对 <code>String</code> 的扩展。
  由于它是一个广泛使用的包，以下 <code>String</code> 属性不会触发此规则，即使它们不是内置的：
</p>
<ul>
  <li> <code>camelize</code> </li>
  <li> <code>capitalize</code> </li>
  <li> <code>classify</code> </li>
  <li> <code>dasherize</code> </li>
  <li> <code>decamelize</code> </li>
  <li> <code>fmt</code> </li>
  <li> <code>loc</code> </li>
  <li> <code>underscore</code> </li>
  <li> <code>w</code> </li>
</ul>


