<p>Internet Explorer offers a way to change the JavaScript code at runtime using conditional comments (activated by a @cc_on statement found in a
comment). Using this preprocessing feature decreases readability and maintainability, and can hinder automated tools. What's more, it is specific to
Internet Explorer and won't work for other browsers.</p>
<p>Most of the time, using those conditional comments can be easily avoided with some refactoring - using modern cross-browsers JavaScript frameworks
and libraries.</p>
<h2>Noncompliant Code Example</h2>
<pre>
/*@cc_on
  @if (@_jscript_version &gt;= 5.5)
    document.write("You are using IE5.5 or newer");
  @else
    document.write("You are using IE5 or older");
  @end
  @*/
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

