<p>
  当<code>for</code>循环的计数器在其增量子句之外递增时，会非常令人困惑。
  在这种情况下，如果可能的话，应该将增量移到循环的增量子句中。
</p>
<h2>Noncompliant Code Example</h2>
<pre>
for (i = 0; i &lt; 10; j++) { // 不合规
  // ...
  i++;
}
</pre>
<h2>合规示例</h2>
<pre>
for (i = 0; i &lt; 10; i++, j++) {
  // ...
}
</pre>
<p>Or </p>
<pre>
for (i = 0; i &lt; 10; i++) {
  // ...
  j++;
}
</pre>

