<p>According to the Backbone.js docs</p>
<blockquote>
  The
  <strong>changed</strong> property is the internal hash containing all the attributes that have changed since the last set. Please do not update
  <strong>changed</strong> directly since its state is internally maintained by set. A copy of
  <strong>changed</strong> can be acquired from changedAttributes.
</blockquote>
<p>The <code>changed</code> property is involved in decisions about whether or not a collection should be resorted when it is updated. If you modify
it manually, you can break the resorting of the collection.</p>
<h2>Noncompliant Code Example</h2>
<pre>
myModel.changed = { myProperty: 1 }; // Non-compliant
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

