<p>While <code>:&lt;element_type&gt;</code> and <code>[type="&lt;element_type&gt;"]</code> can both be used in jQuery to select elements by their
type, <code>[type="&lt;element_type&gt;"]</code> is far faster because it can take advantage of the native DOM <code>querySelectorAll()</code> method
in modern browsers. </p>
<p>This rule raises an issue when following selectors are used:</p>
<ul>
  <li> <code>:checkbox</code> </li>
  <li> <code>:file</code> </li>
  <li> <code>:image</code> </li>
  <li> <code>:password</code> </li>
  <li> <code>:radio</code> </li>
  <li> <code>:reset</code> </li>
  <li> <code>:text</code> </li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
var input = $( "form input:radio" ); // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
var input = $( "form input[type=radio]" ); // Compliant
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

