<p>The <code>yield</code> keyword is used in a generator function to return an <code>IteratorResult</code> to the caller. It has no other purpose, and
if found outside such a function will raise a <code>ReferenceError</code> because it is then treated as an identifier.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function foo() {
  for (var i = 0; i &lt; 5; i++) {
    yield i * 2;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function * foo() {
  for (var i = 0; i &lt; 5; i++) {
    yield i * 2;
  }
}
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

