<p>
  当为键或索引保存值然后无条件覆盖时，可能是开发人员的误操作。
</p>
<h2>Noncompliant Code Example</h2>
<pre>
 fruits[1] = "banana";
 fruits[1] = "apple";  // 不合规 - 索引 1 上的值被覆盖

 myMap.set("key", 1);
 myMap.set("key", 2); // 不合规 - 键“键”的值被替换

 mySet.add(1);
 mySet.add(1); // 不合规 - 元素已经在集合中
</pre>

